### DISCUSSION TITLE
Corrupt ETCD node after underlying ESXI storage failure

### DISCUSSION THREAD

**Discussion by sternma** [etcd-io/etcd#16030] [Q&A]
Has anyone seen this failure in etcd in a Kubernetes environment?

Let me break down our setup:
We have a stacked K8s cluster running "baremetal" on VMs.  These VMs are distributed across various ESXIs that use a shared PureStorage setup.

In one of our clusters, we had a failure in the link from an ESXI to the PS.  In this scenario, all VMs on the affected ESXI immediately failed as they no longer had access to their mounted disk.

When the VMs were migrated to a different ESXI, they all rejoined the cluster successfully.  However, the affected control plane node, which in our stacked setup is also an ETCD node, had constant crashes from ETCD, with symptoms that pointed to a corrupt mount.  Is there any bug or improvement open for this currently, or have others had this issue?

**Comment by ahrtr** [comment]
> an ETCD node, had constant crashes from ETCD, with symptoms that pointed to a corrupt mount

Could you provide more details, e.g the call stack and logs?
